### PR TITLE
refactor: show featured post once

### LINKED ISSUE #6888: Only display hero article once on landing
image
- I think this is not an optimal use of the hero article preview. The article in the hero section should not be displayed in other sections.
- let's make the hero article only appear once on the landing

### PR DESCRIPTION
**Pull request by thnaylor**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

## Context

- [x] Closes #6888
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [ ] Fill up your DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=123PCJXhjb15i6JwVVRGd7KvN2sSNZrtPjr5doJq9oWctoTt)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image

### <samp>🤖 Generated by Copilot at e7f39f5</samp>

Refactored `pages/blog/index.vue` to use a single reduce function for categorizing blog posts by tags. This improves the performance and readability of the blog page component.

### <samp>🤖 Generated by Copilot at e7f39f5</samp>

> _`reduce` function_
> _categorizes blog posts_
> _autumn leaves filter_

### CODE REVIEW

File: pages/blog/index.vue [hunk 1]
```diff
@@ -97,18 +97,31 @@ export default {
...
+      tokens: 'Tokens',
+    }
+
     const latestPosts = imports.sort(
       (a, b) => +new Date(b.attributes.date) - +new Date(a.attributes.date)
     )
 
-    const tokensPosts = latestPosts.filter(
-      (post) => post.attributes.tags === 'Tokens'
+    return latestPosts.reduce(
```
Comment (preschian) [inline pages/blog/index.vue:108]: hhmm, I don't think we need reduce here

just use `shift()` here to remove first array
https://github.com/kodadot/nft-gallery/blob/d09c3b68c076f7214599f1f2de8f32d12f2236c9/pages/blog/index.vue#L70

add more filter here to exclude first array from `latestPosts`
https://github.com/kodadot/nft-gallery/blob/d09c3b68c076f7214599f1f2de8f32d12f2236c9/pages/blog/index.vue#L104-L106

Comment (preschian) [reply pages/blog/index.vue:108]: oh nvm, this is ok 👍

**Review by preschian — APPROVED** [review]
changed my mind, lgtm 👍

File: pages/blog/index.vue [hunk 2]
```diff
@@ -97,18 +97,31 @@ export default {
...
+          acc.posts.push(obj)
+        }
+        return acc
+      },
+      {
+        featured: [],
+        posts: [],
+        tokensPosts: [],
+      }
     )
```
Comment (daiagi) [inline pages/blog/index.vue:124]: 1. change `obj` to `post`
2. change from nested `else if` to early return, makes for a easier read

```suggestion
return latestPosts.reduce(
      (acc, post, index) => {
        if (index === 0) {
          acc.featured.push(post)
          return post
        }  
        if (post.attributes.tags === tags.tokens) {
          acc.tokensPosts.push(post)
          return acc
        } 
          acc.posts.push(obj)
        return acc
      },
      {
        featured: [],
        posts: [],
        tokensPosts: [],
      }
    )
    ```

**Review by daiagi — CHANGES_REQUESTED** [review]

File: pages/blog/index.vue (hunk 2 above)
Comment (thnaylor) [reply pages/blog/index.vue:124]: Thanks will look at the suggestions today and make updates 👍

File: pages/blog/index.vue [hunk 3]
```diff
@@ -97,18 +97,31 @@ export default {
     const resolve = require.context('~/content/blog/', true, /\.md$/)
     const imports = resolve.keys().map((key) => resolve(key))
 
+    const tags = {
+      tokens: 'Tokens',
+    }
```
[the remark this answers is not in the document]
Comment (thnaylor) [reply pages/blog/index.vue:102]: @daiagi I'm not sure if I can do as this file does not use typescript.

**Comment by exezbcz** [comment]
thanks! works for me

File: pages/blog/index.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (daiagi) [reply pages/blog/index.vue:102]: All of our code base is typescript

**Review by daiagi — APPROVED** [review]

**Comment by thnaylor** [comment]
@prury I returned featured posts too early so now it's not adding them to the lastest post. I'll update now. 

As I understand it only the hero section post should be excluded from the 2 sections below.

**Comment by yangwao** [comment]
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 4.48 USD/DOT ~ 6.696 $DOT
🧗 [123PCJXhjb15i6JwVVRGd7KvN2sSNZrtPjr5doJq9oWctoTt](https://polkadot.subscan.io/account/123PCJXhjb15i6JwVVRGd7KvN2sSNZrtPjr5doJq9oWctoTt)
🔗 [0xf4668a6cfbab98e7791386be175d8da28d5296189b24f49988a3b9b07d5e46fc](https://polkadot.subscan.io/extrinsic/0xf4668a6cfbab98e7791386be175d8da28d5296189b24f49988a3b9b07d5e46fc)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**